connector/src/main/scala/com/microsoft/kusto/spark/utils/KustoDataSourceUtils.scala

```diff
@@ -248,35 +253,36 @@ object KustoDataSourceUtils {
       timeout,
       ingestionPropertiesAsJson,
       batchLimit,
-      requestId,
+      sourceParameters.requestId,
       autoCleanupTime,
       minimalExtentsCountForSplitMergePerNode,
       maxRetriesOnMoveExtents,
       adjustSchema
     )
 
-    val sourceParameters = parseSourceParameters(parameters)
-
     if (sourceParameters.kustoCoordinates.table.isEmpty) {
       throw new InvalidParameterException("KUSTO_TABLE parameter is missing. Must provide a destination table name")
     }
 
     logInfo("parseSinkParameters", s"Parsed write options for sink: {'timeout': ${writeOptions.timeout}, 'async': ${writeOptions.isAsync}, " +
       s"'tableCreationMode': ${writeOptions.tableCreateOptions}, 'writeLimit': ${writeOptions.writeResultLimit}, 'batchLimit': ${writeOptions.batchLimit}" +
-      s", 'timeout': ${writeOptions.timeout}, 'timezone': ${writeOptions.timeZone}, 'ingestionProperties': $ingestionPropertiesAsJson, requestId: $requestId}")
+      s", 'timeout': ${writeOptions.timeout}, 'timezone': ${writeOptions.timeZone}, " +
+      s"'ingestionProperties': $ingestionPropertiesAsJson, requestId: ${sourceParameters.requestId}}")
 
     SinkParameters(writeOptions, sourceParameters)
   }
 
-  def getClientRequestProperties(parameters: Map[String, String]): ClientRequestProperties = {
+  def getClientRequestProperties(parameters: Map[String, String], requestId: String): ClientRequestProperties = {
     val crpOption = parameters.get(KustoSourceOptions.KUSTO_CLIENT_REQUEST_PROPERTIES_JSON)
 
-    if (crpOption.isDefined) {
-      val crp = ClientRequestProperties.fromString(crpOption.get)
-      crp
+    val crp = if (crpOption.isDefined) {
+      ClientRequestProperties.fromString(crpOption.get)
     } else {
       new ClientRequestProperties
     }
+
+    crp.setClientRequestId(requestId)
```

Comment:
Assuming the user has provider a requestId, do you allow it or want to override it?

Reply:
if user provided both requestId and ClientRequestProperties the requestId overrides the one in the crp